### PR TITLE
Add initial set of end-to-end credential tests

### PR DESCRIPTION
**Pull request by unexge**
These tests are still WIP just wanted to merge it into our WIP feature branch to trigger CI and get some initial review.

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/e2e-kubernetes/testsuites/credentials.go [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+	}
+
+	ginkgo.Context("Driver level", func() {
+		// TODO: Add tests for current functionality.
+	})
+
+	ginkgo.Context("Pod level", func() {
+		ginkgo.Context("should use correct credentials", func() {
+			ginkgo.It("full access role", func(ctx context.Context) {
+				sa, deleteSA := createServiceAccount(ctx, f, "s3-csi-e2e-sa", annotateServiceAccountWithRole(iamRoleS3FullAccess))
```
Comment (muddyfish) [inline tests/e2e-kubernetes/testsuites/credentials.go:117]: If we're creating service accounts dynamically here, we can probably undo the changes to the cluster generation script

File: tests/e2e-kubernetes/testsuites/credentials.go [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+				defer func() {
+					framework.ExpectNoError(e2epod.DeletePodWithWait(ctx, f.ClientSet, pod))
+				}()
+
+				path := expectWriteToSucceed(pod)
+				expectReadToSucceed(pod, path)
+			})
+
+			ginkgo.It("read-only role", func(ctx context.Context) {
+				sa, deleteSA := createServiceAccount(ctx, f, "s3-csi-e2e-sa", annotateServiceAccountWithRole(iamRoleS3ReadOnlyAccess))
```
Comment (muddyfish) [inline tests/e2e-kubernetes/testsuites/credentials.go:134]: This section of code appears to be duplicated a bunch with minor changes. Would it be possible to refactor it?

File: tests/e2e-kubernetes/testsuites/credentials.go [hunk 3]
```diff
@@ -0,0 +1,256 @@
...
+			l.resources = append(l.resources, resource)
+
+			pod, err := createPodWithServiceAccount(ctx, f.ClientSet, f.Namespace.Name, []*v1.PersistentVolumeClaim{resource.Pvc}, sa.Name)
+			framework.ExpectNoError(err)
+
+			path := expectWriteToSucceed(pod)
+			expectReadToSucceed(pod, path)
+
+			// Associate SA with read-only access role
+			saClient := f.ClientSet.CoreV1().ServiceAccounts(f.Namespace.Name)
```
Comment (muddyfish) [inline tests/e2e-kubernetes/testsuites/credentials.go:171]: Didn't we create a utility function for this?

File: tests/e2e-kubernetes/testsuites/credentials.go [hunk 4]
```diff
@@ -0,0 +1,256 @@
...
+			podReadOnlyAccess, err := createPodWithServiceAccount(ctx, f.ClientSet, f.Namespace.Name, []*v1.PersistentVolumeClaim{resource.Pvc}, saReadOnlyAccess.Name)
+			framework.ExpectNoError(err)
+			defer func() {
+				framework.ExpectNoError(e2epod.DeletePodWithWait(ctx, f.ClientSet, podReadOnlyAccess))
+			}()
+
+			path := expectWriteToSucceed(podFullAccess)
+			expectReadToSucceed(podFullAccess, path)
+
+			expectWriteToFail(podReadOnlyAccess)
```
Comment (muddyfish) [inline tests/e2e-kubernetes/testsuites/credentials.go:239]: Shouldn't this also have a `expectReadToSucceed` for `podReadOnlyAccess`?

File: tests/e2e-kubernetes/testsuites/credentials.go (hunk 2 above)
Comment (unexge) [reply tests/e2e-kubernetes/testsuites/credentials.go:153]: Good point, I'll factor out common parts in follow-up PRs

File: tests/e2e-kubernetes/testsuites/credentials.go (hunk 3 above)
Comment (unexge) [reply tests/e2e-kubernetes/testsuites/credentials.go:191]: There is a utility for creating SA, but this is just creates a client and updates it, we can create a util for this one as well.
